From 51ee75cd516b514a964ede3571205c249e26eec5 Mon Sep 17 00:00:00 2001
From: James Page <james.page@chainguard.dev>
Date: Tue, 12 Aug 2025 15:13:01 +0000
Subject: [PATCH 2/2] Use default KeyStore type in more places.

Set the default keystore type to the runtime configured default rather
than "JKS", supporting frictionless use with other cryptography
providers.
---
 engine/src/main/java/org/apache/hop/www/SslConfiguration.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/engine/src/main/java/org/apache/hop/www/SslConfiguration.java b/engine/src/main/java/org/apache/hop/www/SslConfiguration.java
index 6c2559695d..e928aa5f23 100644
--- a/engine/src/main/java/org/apache/hop/www/SslConfiguration.java
+++ b/engine/src/main/java/org/apache/hop/www/SslConfiguration.java
@@ -16,6 +16,7 @@
  */
 package org.apache.hop.www;
 
+import java.security.KeyStore;
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.Validate;
 import org.apache.hop.core.Const;
@@ -35,7 +36,7 @@ public class SslConfiguration {
   private static final String EMPTY = "empty";
   private static final String NULL = "null";
   private static final String CONST_ILLEGAL_PARAMETER = "WebServer.Error.IllegalSslParameter";
-  @HopMetadataProperty private String keyStoreType = "JKS";
+  @HopMetadataProperty private String keyStoreType = KeyStore.getDefaultType();
   @HopMetadataProperty private String keyStore;
 
   @HopMetadataProperty(password = true)
-- 
2.50.1

